common/component/redis/redis.go

```diff
@@ -252,3 +255,65 @@ type RedisError string
 func (e RedisError) Error() string { return string(e) }
 
 func (RedisError) RedisError() {}
+
+func (s *Settings) refreshTokenRoutineForRedis(ctx context.Context, redisClient RedisClient, version string, meta map[string]string, logger logger.Logger) {
+	ticker := time.NewTicker(tokenRefreshInterval)
+	defer ticker.Stop()
+
+	if !s.useAzureAD {
+		return
+	}
+
+	maxRetries := 3
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			retry := 0
+			for retry < maxRetries {
+				env, err := azure.NewEnvironmentSettings(meta)
+				if err != nil {
+					logger.Error("Failed to get Azure environment settings:", err)
+					retry++
+					continue
+				}
+				tokenCred, err := env.GetTokenCredential()
+				if err != nil {
+					logger.Error("Failed to get Azure AD token credential:", err)
+					retry++
+					time.Sleep(5 * time.Second)
+					continue
+				}
+				at, err := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
+					Scopes: []string{
+						env.Cloud.Services[azure.ServiceOSSRDBMS].Audience + "/.default",
```

Comment:
@ItalyPaleAle  does this `dapr.io/oss-rdbms` environment setting always exist? I don't think so?